File: packages/types/src/event.ts
/**
 * Event types — docs/11_DOMAIN_MODEL.md
 */

export type EventStatus =
  | 'DRAFT'
  | 'CONFIGURATION'
  | 'READY'
  | 'LIVE'
  | 'PAUSED'
  | 'EMERGENCY'
  | 'COMPLETED'
  | 'CANCELLED';

export type GateType = 'ENTRY' | 'EXIT' | 'BIDIRECTIONAL' | 'SMART' | 'EMERGENCY';
export type GateStatus = 'OPEN' | 'CLOSED' | 'RESTRICTED' | 'EMERGENCY_ONLY';
export type RouteType = 'ONE_WAY' | 'TWO_WAY' | 'EMERGENCY' | 'POLICE_ONLY' | 'TEMPORARY';
export type ZoneType = 'GENERAL' | 'VIP' | 'STAGE' | 'FOOD' | 'MEDICAL' | 'ASSEMBLY' | 'RESTRICTED';

export interface GeoPoint {
  lat: number;
  lng: number;
}

export interface Event {
  id: string;
  name: string;
  description: string;
  venue_polygon: GeoPoint[];
  start_date: string;
  end_date: string;
  status: EventStatus;
  owner_id: string;
  expected_attendance: number;
  max_capacity: number;
}

export interface Zone {
  id: string;
  event_id: string;
  name: string;
  polygon: GeoPoint[];
  capacity: number;
  warning_threshold: number;
  critical_threshold: number;
  zone_type: ZoneType;
}

export interface Gate {
  id: string;
  event_id: string;
  zone_id: string;
  name: string;
  location: GeoPoint;
  type: GateType;
  status: GateStatus;
  capacity_per_minute: number;
}

export interface Route {
  id: string;
  event_id: string;
  name: string;
  path: GeoPoint[];
  type: RouteType;
  is_active: boolean;
  capacity: number;
}

File: apps/api/features/events/domain/entities/event.py
"""
Events Feature — Event Entity
The core Event root entity.
"""

from dataclasses import dataclass, field
from shared.domain.base_entity import BaseEntity
from features.events.domain.enums.event_status import EventStatus
from features.events.domain.value_objects.geo_point import GeoPoint
from features.events.domain.value_objects.date_range import DateRange


@dataclass
class Event(BaseEntity):
    name: str = ""
    description: str = ""
    venue_polygon: list[GeoPoint] = field(default_factory=list)
    date_range: DateRange = field(default_factory=lambda: DateRange(None, None))
    status: EventStatus = EventStatus.DRAFT
    owner_id: str = ""
    expected_attendance: int = 0
    max_capacity: int = 0
